src/node/internal/internal_net.ts

```diff
@@ -1464,11 +1496,12 @@ function addClientAbortSignalOption(self: Socket, signal: AbortSignal): void {
   function onAbort(): void {
     disposable?.[Symbol.dispose]();
     self._aborted = true;
-    // TODO(now): What else should be do here? Anything?
   }
 
   if (signal.aborted) {
-    onAbort();
+    queueMicrotask(() => {
+      onAbort();
+    });
```

Comment:
This is the actual fix for the abortsignal errors. it should not be thrown synchronously